incubator/azuremonitor-containers/templates/omsagent-daemonset.yaml

```diff
@@ -24,8 +24,12 @@ spec:
    serviceAccountName: omsagent
    {{- end }}
    containers:
-     - name: omsagent 
+     - name: omsagent
+       {{- if eq .Values.omsagent.domain "opinsights.azure.cn" }}
```

Comment:
since this is free form input from user, downcase & trim (beginning & end) white spaces